src/test/java/com/example/usersapi/controller/UsersApiControllerTest.java

```diff
@@ -0,0 +1,268 @@
+import com.example.usersapi.service.UserServiceImpl;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.InjectMocks;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.EnableAutoConfiguration;
+import org.springframework.boot.autoconfigure.security.servlet.SecurityAutoConfiguration;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.http.MediaType;
+import org.springframework.security.test.context.support.WithMockUser;
+import org.springframework.test.context.junit4.SpringRunner;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.MvcResult;
+import org.springframework.test.web.servlet.RequestBuilder;
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.mockito.Mockito.when;
+
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.content;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@RunWith(SpringRunner.class)
+@SpringBootTest
+@AutoConfigureMockMvc
+@EnableAutoConfiguration(exclude = { SecurityAutoConfiguration.class})
+public class UsersApiControllerTest {
+
+    @Autowired
+    private MockMvc mockMvc;
+
+    @MockBean
+    private UserServiceImpl userService;
+
+    @MockBean
+    private UserProfileServiceImpl userProfileService;
+
+    @MockBean
+    private PostClient postClient;
+
+    @MockBean
+    private CommentClient commentClient;
+
+    @InjectMocks
+    private JwtResponse jwtResponse;
+
+    @InjectMocks
+    private User user;
+
+    @InjectMocks
+    private UserRole userRole;
+
+    @InjectMocks
+    private UserProfile userProfile;
+
+    @InjectMocks
+    private PostBean postBean;
+
+    @InjectMocks
+    private CommentBean commentBean;
+
+    private List<User> userList;
+
+    @Before
+    public void init() {
+        jwtResponse.setToken("12345");
+        jwtResponse.setId(1);
+        jwtResponse.setUsername("testUser");
+
+        userRole.setId(3);
+        userRole.setName("ROLE_USER");
+
+        userProfile.setId(1);
+        userProfile.setUserId(1);
+        userProfile.setAdditionalEmail("additionalEmail@email.com");
+        userProfile.setAddress("123 Test St.");
+        userProfile.setMobile("1-800-TEST-NUM");
+
+        user.setId(1);
+        user.setUsername("testUser");
+        user.setPassword("testPass");
+        user.setEmail("user@testmail.com");
+        user.setUserRole(userRole);
+        user.setUserProfile(userProfile);
+
+        userList = new ArrayList<>();
+        userList.add(user);
+    }
+
+    @Test
+    public void signup_ReturnsJwtResponse_Success() throws Exception {
```

Comment:
Does this matter that we're catching general Exceptions?  Is this a place holder?  I assume it doesn't really matter since everything bubbles up anyway.